### PR TITLE
add gcp enclave page V2.

### PR DESCRIPTION
**Pull request by yishi-ttd**
Add a new page for GCP enclave v2 (OIDC).
 - Mimiced existing GCP enclave page.

Screenshot:
image

### CODE REVIEW

File: webroot/adm/enclave-gcp-v2.html [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+      console.log("enclaveId:" + enclaveId);
+      $('#enclaveId').text(enclaveId);
+    });
+
+    // update gcloud Cli
+    const imageFamily = isDebug ? 'confidential-space-debug' : 'confidential-space';
+    const cli = gcloudCli
+            .replaceAll('<IMAGE_FAMILY>', imageFamily)
+            .replaceAll('<IMAGE_SHA>', imageSha)
+            .replaceAll('<API_TOKEN>', apiToken)
```
Comment (thomasm-ttd) [inline webroot/adm/enclave-gcp-v2.html:166]: The API token is considered sensitive and we never send it via email, or in any text files. So the resulting gcloud CLI command must still have <API_TOKEN> (the actual string, not a replaced value). As the API key isn't used for the generation of the enclave Id, there is no reason for it to be collected on this page.

File: webroot/adm/enclave-gcp-v2.html [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+<h1>UID2 Admin - GCP Enclave ID Tool(V2)</h1>
+
+<a href="/">Back</a>
+
+<br>
+<br>
+
+<h3>Inputs</h3>
+
+<label for="apiToken" class="bold">Operator Api Key:</label>
```
Comment (thomasm-ttd) [inline webroot/adm/enclave-gcp-v2.html:23]: We should not be collecting the API key here - we don't need it for the enclave Id, so we should not collect it. It is considered to be a secret.

Comment (yishi-ttd) [reply webroot/adm/enclave-gcp-v2.html]: ACK, will update.

File: webroot/adm/enclave-gcp-v2.html (hunk 1 above)
Comment (yishi-ttd) [reply webroot/adm/enclave-gcp-v2.html]: ACK, will update.

File: webroot/adm/enclave-gcp-v2.html [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+<br>
+
+<label for="enclaveId">Generated GCP Enclave ID:</label>
+<br>
+<pre><label id="enclaveId">TBD</label></pre>
+<br>
+<br>
+
+<script language="JavaScript">
+  const gcloudCli = '$ gcloud compute instances create <INSTANCE_NAME> \\\n\
```
Comment (lunwang-ttd) [inline webroot/adm/enclave-gcp-v2.html:86]: Should we add a placeholder for zone config so that partners can change it?

Comment (lunwang-ttd) [reply webroot/adm/enclave-gcp-v2.html:86]: Also for `machine-type`

Comment (yishi-ttd) [reply webroot/adm/enclave-gcp-v2.html:77]: There would be too many GCP overrides. I would suggest to only provide necessary commands for partners. i.e. they will use default zone & machine-type if not specified, but OK to change. We could mention this in public doc guide.
WDYT?

Comment (thomasm-ttd) [reply webroot/adm/enclave-gcp-v2.html:77]: I agree, we can document what else they can change

**Review by thomasm-ttd — APPROVED** [review]

File: webroot/adm/enclave-gcp-v2.html (hunk 3 above)
Comment (lunwang-ttd) [reply webroot/adm/enclave-gcp-v2.html:77]: I see, a well written doc will be good enough

**Review by lunwang-ttd — APPROVED** [review]

**Pull request merged**